Здравствуйте!

Правила проведения контрольной работы: пользоваться можно чем угодно, но общаться ни с кем не надо – за это может быть штраф. И за выбор не своего варианта может быть штраф.

Контрольная работа проходит в [webinar.ru](webinar.ru) в общем зале. Преподаватель может попросить включить звук и демонстрациию всего экрана **(а не какого-то отдельного окна – выбирайте левую верхнюю опцию)**. Поэтому постарайтесь найти нешумное место на время контрольной работы. Организационные и прочие уместные **вопросы задавайте письменно в чате, а не голосом**; за вопросы голосом в общем зале во время работы может быть штраф. Собственные наушники|аудиоколонки поставьте на минимальную громкость, чтобы случайные посторонние звуки в [webinar.ru](webinar.ru) не отвлекали вас.

Вам потребуется один-два ключа к API YouTube. Если ключей не хватит, обратитесь к преподавателю письменно в чате.

Также вам самим следует понять, предполагает ли поставленная задача обращение к прилагаемым базам с видео и с каналами.

Если Вы встретитесь с **непреодолимыми затруднениями** при решении какой-либо задачи, можете обратиться письменно в чате к преподавателю за **подсказкой**. Но каждая подсказка предполагает штраф (2 балла из 10). Также можете попробовать **написать код для задач, следующих за нерешённой задачей**. Если он окажется верен, то получите за него какую-то часть балла.

Запустите следующий чанк для получения задач. Решайте задачи прямо в **этом** скрипте. Выполненную работу (этот скрипт с решёнными задачами и -- опционально -- итоговый файл формата .xlsx ) отправьте на адрес [task.hse.1@gmail.com](task.hse.1@gmail.com) не позднее 90 минут с начала работы.

In [3]:
variant = input('Впишите свои фамилию и имя: ')
import pandas
variantS_df = pandas.read_excel('Варианты.xlsx', index_col=0)
variantS_df = variantS_df.dropna()
variant_df = variantS_df[variantS_df['ФИО'].str.contains(variant, case=False)]
if len(variant_df) == 0:
    print("Вы не найдены в списке студентов. Попробуйте перезапустить чанк и заново вписать свои фамилию и имя."
          , "Если это не поможет, обратитесь к преподавателю письменно в чате."
          , sep='\n')
else:
    display(variant_df.drop(variant_df.columns[-1], axis=1))
    variant = int(input('Это Вы? Для подтверждения впишите НОМЕР СТРОКИ, в которой видите Ваши ФИО: '))
    print(f"***"
          , f"Здравствуйте, {variant_df['ФИО'][variant].iloc[0]}!"
          , f"***"
          , f"{variant_df.columns[-1]}"
          , f"***"
          , f"{variant_df[variant_df.columns[-1]][variant].iloc[0]}"
          , f"***"
          , f"{variant_df[variant_df.columns[-1]][variant].iloc[1]}"
          , f"***"
          , f"{variantS_df.iloc[-6, -1]}"
          , f"***"
          , f"{variantS_df.iloc[-5, -1]}"
          , f"***"
          , f"{variantS_df.iloc[-4, -1]}"
          , f"***"
          , f"{variantS_df.iloc[-3, -1]}"
          , f"***"
          , f"{variantS_df.iloc[-2, -1]}"
          , f"***"
          , f"{variantS_df.iloc[-1, -1]}"
          , f"***"
          , f"Успеха!"
          , sep='\n')

ФИО
16  Казачинский Андрей Андреевич
16  Казачинский Андрей Андреевич

***
Здравствуйте, Казачинский Андрей Андреевич!
***
Задача 0. Загрузите обе базы данных из файлов Excel папки КР 2. Прочтите следующие задачи и выясните, какая из двух баз содержит данные, релевантные для решения следующих задач. Обоснуйте Ваш выбор демонстрацией посредством кода наличия в выбранной базе требуемых переменных.
***
Задача 1. В релевантной базе данных отберите только объекты, которые удовлетворяют условию: snippet.country не указано (NaN).
***
Задача 2. Вам предлагаются следующие переменные из базы: Число_категорий_в_topicDetails.topicCategories и Число_слов_в_brandingSettings.channel.keywords. Дайте письменное описание этих переменных (их нет в релевантной базе, но из их названий ясно, какие переменные следует взять из базы в качестве исходных и как их преобразовать) в терминах типа шкалы переменной, насколько распределение однородно, симметрично или смещено, много ли выбросов. Описание должно включать релевантные диаграммы и величины релевантных статистик.
***
Задача 3.

## Задача 0.
## Загрузите обе базы данных из файлов Excel папки КР 2. Прочтите следующие задачи и выясните, какая из двух баз содержит данные, релевантные для решения следующих задач. Обоснуйте Ваш выбор демонстрацией посредством кода наличия в выбранной базе
требуемых переменных.

In [8]:
# 1
import pandas as pd

df_1 = pd.read_excel('База_0.xlsx')
df_1.head()

Unnamed: 0             kind                         etag  \
0           1  youtube#channel  e87chamF3P_n5wVY29muRfK5tIo   
1           2  youtube#channel  UTK2MoWzSVZk3ZOl4Mp4wvVUsu0   
2           3  youtube#channel  DrIibs1d8Qp3S9euGILr_5ablEk   
3           4  youtube#channel  nZbZAE3Rx7QtFHx9enbgxUj7c2g   
4           5  youtube#channel  G3e8sxvPSp6IftL1Tc88OrsQ3BY   

                         id        snippet.title  \
0  UCQ7HpkBdKFirV0IYdqKiRxg         Edwords Live   
1  UCrverqq03KW1i8AXWdDqbjA  Павел Александрович   
2  UCmmxjr5Bz2Zws7Ov4-VL6Ww          ТелеРеклама   
3  UCGkj1MiJiLkcWogLghl2Cuw        Right Puperzz   
4  UCzj0pC6F8vK8bSrJNQrPMww          All Reklama   

                                 snippet.description snippet.customUrl  \
0  Это мой лайв канал. Здесь я выкладываю новости...       edwordslive   
1                                                NaN               NaN   
2                                                NaN               NaN   
3              Я делаю смешные пупы, подписывайся :D               NaN   
4  На канале собраны интересные рекламные ролики ...        allreklama   

           snippet.publishedAt  \
0         2012-03-29T20:15:38Z   
1         2014-04-28T22:56:43Z   
2  2020-03-08T07:05:12.456512Z   
3         2016-08-08T23:38:10Z   
4         2014-02-07T14:26:06Z   

                      snippet.thumbnails.default.url  \
0  https://yt3.ggpht.com/ytc/AKedOLT_sFEMP-ZNu4US...   
1  https://yt3.ggpht.com/ytc/AKedOLSj2mk4DA1mfZrB...   
2  https://yt3.ggpht.com/ytc/AKedOLQr1oSpb4mA0IIr...   
3  https://yt3.ggpht.com/ytc/AKedOLTbV86DFVKosM5A...   
4  https://yt3.ggpht.com/ytc/AKedOLSRG9g5x4hCKqv-...   

   snippet.thumbnails.default.width  ...  localizations.ja_JP.description  \
0                                88  ...                              NaN   
1                                88  ...                              NaN   
2                                88  ...                              NaN   
3                                88  ...                              NaN   
4                                88  ...                              NaN   

  localizations.cs_CZ.title  localizations.cs_CZ.description  \
0                       NaN                              NaN   
1                       NaN                              NaN   
2                       NaN                              NaN   
3                       NaN                              NaN   
4                       NaN                              NaN   

   localizations.zh_HK.title localizations.zh_HK.description  \
0                        NaN                             NaN   
1                        NaN                             NaN   
2                        NaN                             NaN   
3                        NaN                             NaN   
4                        NaN                             NaN   

   localizations.es_ES.title  localizations.it_IT.title  \
0                        NaN                        NaN   
1                        NaN                        NaN   
2                        NaN                        NaN   
3                        NaN                        NaN   
4                        NaN                        NaN   

  localizations.it_IT.description localizations.es_US.title  \
0                             NaN                       NaN   
1                             NaN                       NaN   
2                             NaN                       NaN   
3                             NaN                       NaN   
4                             NaN                       NaN   

  localizations.es_US.description  
0                             NaN  
1                             NaN  
2                             NaN  
3                             NaN  
4                             NaN  

[5 rows x 260 columns]

In [9]:
# 2
df_2 = pd.read_excel('База_1.xlsx')

df_1.head()

Unnamed: 0             kind                         etag  \
0           1  youtube#channel  e87chamF3P_n5wVY29muRfK5tIo   
1           2  youtube#channel  UTK2MoWzSVZk3ZOl4Mp4wvVUsu0   
2           3  youtube#channel  DrIibs1d8Qp3S9euGILr_5ablEk   
3           4  youtube#channel  nZbZAE3Rx7QtFHx9enbgxUj7c2g   
4           5  youtube#channel  G3e8sxvPSp6IftL1Tc88OrsQ3BY   

                         id        snippet.title  \
0  UCQ7HpkBdKFirV0IYdqKiRxg         Edwords Live   
1  UCrverqq03KW1i8AXWdDqbjA  Павел Александрович   
2  UCmmxjr5Bz2Zws7Ov4-VL6Ww          ТелеРеклама   
3  UCGkj1MiJiLkcWogLghl2Cuw        Right Puperzz   
4  UCzj0pC6F8vK8bSrJNQrPMww          All Reklama   

                                 snippet.description snippet.customUrl  \
0  Это мой лайв канал. Здесь я выкладываю новости...       edwordslive   
1                                                NaN               NaN   
2                                                NaN               NaN   
3              Я делаю смешные пупы, подписывайся :D               NaN   
4  На канале собраны интересные рекламные ролики ...        allreklama   

           snippet.publishedAt  \
0         2012-03-29T20:15:38Z   
1         2014-04-28T22:56:43Z   
2  2020-03-08T07:05:12.456512Z   
3         2016-08-08T23:38:10Z   
4         2014-02-07T14:26:06Z   

                      snippet.thumbnails.default.url  \
0  https://yt3.ggpht.com/ytc/AKedOLT_sFEMP-ZNu4US...   
1  https://yt3.ggpht.com/ytc/AKedOLSj2mk4DA1mfZrB...   
2  https://yt3.ggpht.com/ytc/AKedOLQr1oSpb4mA0IIr...   
3  https://yt3.ggpht.com/ytc/AKedOLTbV86DFVKosM5A...   
4  https://yt3.ggpht.com/ytc/AKedOLSRG9g5x4hCKqv-...   

   snippet.thumbnails.default.width  ...  localizations.ja_JP.description  \
0                                88  ...                              NaN   
1                                88  ...                              NaN   
2                                88  ...                              NaN   
3                                88  ...                              NaN   
4                                88  ...                              NaN   

  localizations.cs_CZ.title  localizations.cs_CZ.description  \
0                       NaN                              NaN   
1                       NaN                              NaN   
2                       NaN                              NaN   
3                       NaN                              NaN   
4                       NaN                              NaN   

   localizations.zh_HK.title localizations.zh_HK.description  \
0                        NaN                             NaN   
1                        NaN                             NaN   
2                        NaN                             NaN   
3                        NaN                             NaN   
4                        NaN                             NaN   

   localizations.es_ES.title  localizations.it_IT.title  \
0                        NaN                        NaN   
1                        NaN                        NaN   
2                        NaN                        NaN   
3                        NaN                        NaN   
4                        NaN                        NaN   

  localizations.it_IT.description localizations.es_US.title  \
0                             NaN                       NaN   
1                             NaN                       NaN   
2                             NaN                       NaN   
3                             NaN                       NaN   
4                             NaN                       NaN   

  localizations.es_US.description  
0                             NaN  
1                             NaN  
2                             NaN  
3                             NaN  
4                             NaN  

[5 rows x 260 columns]

In [12]:
## 3 Проверка

columns_set_1 = set(df_1.columns)
columns_set_2 = set(df_2.columns)

target_cols = {'topicDetails.topicCategories',  # Переменная `topicDetails.topicCategories` - ключевые слова в описании канала. Тип - строка
               'brandingSettings.channel.keywords' # Переменная `channel.keywords` - ключевые слова в описании канала. Тип - строка
               }

print('Пересечение в df_1', target_cols & columns_set_1)
print('Пересечение в df_2', target_cols & columns_set_2)


Пересечение в df_1 {'brandingSettings.channel.keywords', 'topicDetails.topicCategories'}
Пересечение в df_2 {'topicDetails.topicCategories'}


In [15]:
# 4
# В первом датасете есть целевые столбцы (векторы с данными)

data = df_1

data[list(target_cols)]

/var/folders/s5/4r_7wwf925g2bnv4060z0x400000gn/T/ipykernel_10286/658078458.py:6: FutureWarning: Passing a set as an indexer is deprecated and will raise in a future version. Use a list instead.
  data[target_cols]


brandingSettings.channel.keywords  \
0                                             live лайв   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4     Подборки реклама снг россия украина телевидени...   
...                                                 ...   
1148  Bassivity Digital balkan "hip hop" urban coby ...   
1149  "Павел Воля" "Pavel Volya" "Паша Воля" Воля "C...   
1150              TVN tvn.pl player player.pl telewizja   
1151  "россия 1" россия "россия ютуб" "россия телека...   
1152  Дорн beverly "dorn ivan" дорн "ivan dorn" теле...   

                           topicDetails.topicCategories  
0     ['https://en.wikipedia.org/wiki/Video_game_cul...  
1     ['https://en.wikipedia.org/wiki/Knowledge', 'h...  
2       ['https://en.wikipedia.org/wiki/Entertainment']  
3       ['https://en.wikipedia.org/wiki/Entertainment']  
4       ['https://en.wikipedia.org/wiki/Entertainment']  
...                                                 ...  
1148  ['https://en.wikipedia.org/wiki/Hip_hop_music'...  
1149  ['https://en.wikipedia.org/wiki/Pop_music', 'h...  
1150  ['https://en.wikipedia.org/wiki/Television_pro...  
1151  ['https://en.wikipedia.org/wiki/Society', 'htt...  
1152  ['https://en.wikipedia.org/wiki/Pop_music', 'h...  

[1153 rows x 2 columns]

## Задача 2. Вам предлагаются следующие переменные из базы: Число_категорий_в_topicDetails.topicCategories и Число_слов_в_brandingSettings.channel.keywords.

Дайте письменное описание этих переменных (их нет в релевантной базе, но из их названий ясно, какие переменные следует взять из базы в качестве исходных и как их преобразовать) в терминах типа шкалы переменной, насколько распределение однородно, симметрично или смещено, много ли выбросов. Описание должно включать релевантные диаграммы и величины релевантных статистик.

Исходные переменные
- Переменная `topicDetails.topicCategories` - ключевые слова в описании канала.
    1) Тип - строка.
    2) Должна быть преобразована в список со строчкам, откуда можно посчитать кол-во элементов

- Переменная `channel.keywords` - ключевые слова в описании канала.
    1) Тип - строка.
    2) Должна быть преобразована в список со строчками, откуда можно посчитать кол-во элементов



In [27]:
# 5 Преобразование переменных
import numpy as np

def read_list_from_string(text: str):
    """Функция для преобразования строчки с закодированым массивом внутри обратно в список"""
    text = text[1:-1]
    arr = text.split(',')
    return [el.strip() for el in arr]

# Стратегия

data['Listed_topicDetails.topicCategories'] = data['topicDetails.topicCategories'].apply(lambda x: read_list_from_string(x) if isinstance(x, str) else np.nan)

data['Число_категорий_в_topicDetails.topicCategories'] = data['Listed_topicDetails.topicCategories'].apply(lambda x: len(x) if isinstance(x, list) else np.nan)


data['Число_слов_в_brandingSettings.channel.keywords'] = data['brandingSettings.channel.keywords'].apply(lambda x: len(x.split()) if isinstance(x, str) else np.nan)

display(data[['topicDetails.topicCategories', 'Listed_topicDetails.topicCategories', 'Число_категорий_в_topicDetails.topicCategories', 'brandingSettings.channel.keywords', 'Число_слов_в_brandingSettings.channel.keywords']].sample(10))

topicDetails.topicCategories  \
111                                                NaN   
178                                                NaN   
837  ['https://en.wikipedia.org/wiki/Lifestyle_(soc...   
76   ['https://en.wikipedia.org/wiki/Technology', '...   
735            ['https://en.wikipedia.org/wiki/Music']   
339  ['https://en.wikipedia.org/wiki/Lifestyle_(soc...   
50   ['https://en.wikipedia.org/wiki/Technology', '...   
441            ['https://en.wikipedia.org/wiki/Music']   
377  ['https://en.wikipedia.org/wiki/Entertainment'...   
495  ['https://en.wikipedia.org/wiki/Entertainment'...   

                   Listed_topicDetails.topicCategories  \
111                                                NaN   
178                                                NaN   
837  ['https://en.wikipedia.org/wiki/Lifestyle_(soc...   
76   ['https://en.wikipedia.org/wiki/Technology', '...   
735            ['https://en.wikipedia.org/wiki/Music']   
339  ['https://en.wikipedia.org/wiki/Lifestyle_(soc...   
50   ['https://en.wikipedia.org/wiki/Technology', '...   
441            ['https://en.wikipedia.org/wiki/Music']   
377  ['https://en.wikipedia.org/wiki/Entertainment'...   
495  ['https://en.wikipedia.org/wiki/Entertainment'...   

     Число_категорий_в_topicDetails.topicCategories  \
111                                             NaN   
178                                             NaN   
837                                             2.0   
76                                              2.0   
735                                             1.0   
339                                             3.0   
50                                              4.0   
441                                             1.0   
377                                             4.0   
495                                             2.0   

                     brandingSettings.channel.keywords  \
111                                                NaN   
178                                                NaN   
837                                                NaN   
76                                                 NaN   
735                                                NaN   
339                                                NaN   
50   rytp inex cenine RYTP InEx IpUp "Правильная ре...   
441                                                NaN   
377  salomatin саломатин sergey сергей куинджи любо...   
495                                                NaN   

     Число_слов_в_brandingSettings.channel.keywords  
111                                             NaN  
178                                             NaN  
837                                             NaN  
76                                              NaN  
735                                             NaN  
339                                             NaN  
50                                             14.0  
441                                             NaN  
377                                            22.0  
495                                             NaN

**Ключевые переменные теперь представлены в формате числовых переменных**

In [28]:
# 6 Оценка переменной 'Число_категорий_в_topicDetails.topicCategories' & 'Число_слов_в_brandingSettings.channel.keywords'


from randan.descriptive_statistics import ScaleStatistics


vars = ['Число_категорий_в_topicDetails.topicCategories',
        'Число_слов_в_brandingSettings.channel.keywords'
        ]


ss = ScaleStatistics(data, vars,
                     normality_test=True)




SCALE STATISTICS SUMMARY
------------------



/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/descriptive_statistics.py:491: FutureWarning: this method is deprecated in favour of `Styler.format(precision=..)`
  .set_precision(n_decimals))


Following variables have multiple modes: Число_слов_в_brandingSettings.channel.keywords

NORMALITY TESTS
------------------



/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/descriptive_statistics.py:460: FutureWarning: this method is deprecated in favour of `Styler.format(precision=..)`
  .set_precision(n_decimals))


In [39]:
# 7 Графическая оценка распределения 'Число_категорий_в_topicDetails.topicCategories'

print(vars[0])

# 10 График распределения statistics.viewCount
import plotly.express as px

fig = px.histogram(data, x=data[vars[0]],
                   marginal='box',
                   nbins=10)
fig.show()

Число_категорий_в_topicDetails.topicCategories


In [38]:
# 7 Графическая оценка распределения 'Число_слов_в_brandingSettings.channel.keywords'

print(vars[1])

# 10 График распределения statistics.viewCount
import plotly.express as px

fig = px.histogram(data, x=data[vars[1]],
                   marginal='box',
                   nbins=100)
fig.show()

Число_слов_в_brandingSettings.channel.keywords


### Резюме по переменным

**Переменная `Число_категорий_в_topicDetails.topicCategories`**

Интервальная шкала

Показатель нормальности - 0.251 (p-value < 0.05)
Есть отдельные выбросы, центр распределения смещен в сторону 2, однако есть четкий пик

Распределение данной переменной близко к нормальному


**Переменная `Число_слов_в_brandingSettings.channel.keywords`**

Интервальная шкала

Показатель нормальности - 0.161 (p-value < 0.05)
Есть выбросы, центр распределения смещен в сторону 17, однако есть несколько пиков (в точках 2, 4, 6, 10)

Распределение не близко к нормальному

## Задача 3. Постарайтесь письменно сформулировать для них содержательную гипотезу в терминах линейной связи: её отсутствия | наличия, направления и силы. Также письменно сформулируйте релевантную статистическую гипотезу.


**Содержательная гипотеза:**

В каналах с большим числом категорий - большее число ключевых слов

**Статистическая гипотеза:**

H0: Линейной связи между признаками нет
H1: Линейная связь между признаками есть

## Задача 4. Проверьте эти гипотезы графически и математически; проинтерпретируйте результат.

In [40]:
# 8 Графическая оценка связей между переменными


fig = px.scatter(data, x=vars[0], y=vars[1], trendline='ols')

fig.show()

In [41]:
# 13 Проверка на линейную связь между признаками

from randan.bivariate_association import Correlation

corr = Correlation(data, vars, show_results=False)
corr.show_results()


CORRELATION SUMMARY (PEARSON METHOD, 2-TAILED)
------------------


/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/bivariate_association.py:397: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`



Note: Each empty index duplicates the previous one.
Maximum correlation is 0.024 (p-value 0.63) for variables Число_категорий_в_topicDetails.topicCategories and Число_слов_в_brandingSettings.channel.keywords,
minimum correlation is 0.024 (p-value 0.63) for variables Число_категорий_в_topicDetails.topicCategories and Число_слов_в_brandingSettings.channel.keywords.


### Интерпретация линейной связи

Коээфициент корреляции - 0.024, p-value = 0.630 (0.63 > 0.05)

Таким образом, мы принимаем ***гипотезу H0 об отсутствии линейной связи между признаками в генеральной совокупности*** на уровне 95%

Соответственно, линейной связи нет, возможно, связь имеется, но они нелинейная и/или не монотонна

## Задача 5. Постарайтесь письменно сформулировать для них содержательную гипотезу в терминах КРИВОлинейной связи: её наличия|отсутствия в целом и между отдельными категориями. Также письменно сформулируйте релевантную статистическую гипотезу.


In [53]:
# 14 Проверка на аппроксимацию с помощью кривой
fig = px.scatter(data, x=vars[0], y=vars[1], trendline='lowess')

fig.show()

Мы пришли к тому, что  между признаками отсутствуте линейная связь. А также графически криволинейная связь не подтверждается (аппроксимированная кривая по Lowess не выделяет пиков средних значений)

Соответственно, мы можем проверить наличие связи между группами с помощью кросс-таблиц и $\chi ^{2} $

**Содержательная гипотеза:**

Каналы имеющие разное количество заявленных тематик имеют разное

**Статистическая гипотеза:**


H0: $\chi ^{2}  = 0 $ в ГС
H1: $\chi^{2}  != 0 $ в ГС

In [45]:
# 14 формируем группы для создания таблицы и проверки связи
inv_borders = [
                       np.nanpercentile(data['Число_слов_в_brandingSettings.channel.keywords'], 25),
                       np.nanpercentile(data['Число_слов_в_brandingSettings.channel.keywords'], 50),
                       np.nanpercentile(data['Число_слов_в_brandingSettings.channel.keywords'], 75),
                       np.nanpercentile(data['Число_слов_в_brandingSettings.channel.keywords'], 100)]


def make_group_from_borders(word_count):
    if word_count < inv_borders[0]:
        return '1 Low word_count'
    elif inv_borders[0] <= word_count < inv_borders[1]:
        return '2 Lower medium word_count'
    elif inv_borders[1] <= word_count < inv_borders[2]:
        return '3 Higher medium word_count'
    else:
        return '4 High word_count'

data['word_count.group'] = data['Число_слов_в_brandingSettings.channel.keywords'].apply(make_group_from_borders)

In [47]:
# 16 выводим статистику по группам

from randan.descriptive_statistics import NominalStatistics

var = 'word_count.group'

ns = NominalStatistics(data, [var])

freq_table = ns.frequencies()[var]


NOMINAL STATISTICS SUMMARY
------------------



/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/descriptive_statistics.py:72: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`




FREQUENCIES
------------------

variable: word_count.group


/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/descriptive_statistics.py:93: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`



## Задача 6. Проверьте эти гипотезы математически. Строго соблюдайте ограничения применяемых методов. Проинтерпретируйте результат.

In [49]:
# 16 кросстаблица
from randan.bivariate_association import Crosstab

ctab = Crosstab(data, 'Число_категорий_в_topicDetails.topicCategories', 'word_count.group')


CROSSTAB SUMMARY
------------------

Observed frequencies


/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/bivariate_association.py:238: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`



------------------

Expected frequencies


/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/bivariate_association.py:244: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`



11 (39.286%) cells have expected frequency less than 5. The minimum expected frequency is 0.09.
------------------

Chi-square statistic is 21.511 (p-value = 0.254).
------------------

Pearson's residuals


/Users/andrey/PycharmProjects/DataMiningProjects/.venv/lib/python3.10/site-packages/randan/bivariate_association.py:253: FutureWarning:

this method is deprecated in favour of `Styler.format(precision=..)`



1 (3.571%) cells have Pearson's residual bigger than 1.96. 
The biggest residual is 2.829 (categories 7.0 and 2 Lower medium word_count).
The smallest residual is -0.834 (categories 7.0 and 4 High word_count).


In [52]:
# 17 Оценка значимости
print('P-value', ctab.pvalue)
print('Level of Significance', ctab.sig_level)

ctab.pvalue < ctab.sig_level

P-value 0.25442284023410583
Level of Significance 0.05


False

### Интерпретация

Основываясь на данных теста и наблюдаемых частотах, мы не имеем оснований отклонить нулевую гипотезу (p-value > level of significance)

Таким образом, мы не можем принять альтернативную гипотезу H1 о наличии связи между количеством категорий и среднем количестве ключевых слов в ГС